File: packages/mlmpfr.4.0.2+dune/testsuite/conversion/conversionfunctions.c
/* This file is part of mlmpfr.

  mlmpfr is free software: you can redistribute it and/or modify it
  under the terms of the GNU Lesser General Public License as
  published by the Free Software Foundation, either version 3 of the
  License, or (at your option) any later version.

  mlmpfr is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
  Lesser General Public License for more details.

  You should have received a copy of the GNU Lesser General Public
  License along with mlmpfr. If not, see
 <http://www.gnu.org/licenses/>. */

#include <stdlib.h>
#include <stdio.h>
#include <mpfr.h>

int main(int argc, char **argv)
{
  mpfr_t x, y, z;
  double d;
  int i;
  mpfr_exp_t e;

  if (argc != 2)
    exit (1);

  mpfr_init (x);
  mpfr_set_d (x, 1./3., MPFR_RNDN);

  d = mpfr_get_d (x, MPFR_RNDN);
  printf ("%e\n", d);

  i = mpfr_get_si (x, MPFR_RNDN);
  printf ("%d\n", i);

  d = mpfr_get_d_2exp (&e, x, MPFR_RNDN);
  printf("%e %ld\n", d, e);

  mpfr_init (y);
  mpfr_frexp (&e, y, x, MPFR_RNDN);
  printf("%ld\n", e);

  char *buffer;
  buffer = mpfr_get_str (NULL, &e, 10, 0, x, MPFR_RNDN);
  printf("%s %ld\n", buffer, e);

  mpfr_printf ("%.Re\n", x);

  FILE *stream;
  stream = fopen ("fpif_export.expected", "w");
  mpfr_fpif_export (stream, x);
  fclose (stream);

  mpfr_init (z);
  stream = fopen (argv[1], "r");
  mpfr_fpif_import (z, stream);
  mpfr_printf ("%.Re\n", z);
  fclose (stream);

  if (mpfr_fits_sint_p (x, MPFR_RNDN))
    printf ("true\n");
  else
    printf ("false\n");

  mpfr_free_str (buffer);
  mpfr_clear (x);
  mpfr_clear (y);
  return 0;
}

File: packages/mlmpfr.4.1.0/testsuite/conversion/conversionfunctions.ml
(* This file is part of mlmpfr.

  mlmpfr is free software: you can redistribute it and/or modify it
  under the terms of the GNU Lesser General Public License as
  published by the Free Software Foundation, either version 3 of the
  License, or (at your option) any later version.

  mlmpfr is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
  Lesser General Public License for more details.

  You should have received a copy of the GNU Lesser General Public
  License along with mlmpfr. If not, see
  <http://www.gnu.org/licenses/>. *)

open Printf
module M = Mpfr

let _ =
  let x = M.make_from_float (1. /. 3.) in
  printf "%e\n" (M.get_float x);
  printf "%d\n" (M.get_int x);
  let m, n = M.get_float_2exp x in
  printf "%e %d\n" m n;
  let _, n = M.get_mpfr_2exp x in
  printf "%d\n" n;
  printf "%d\n" (M.get_str_ndigits 10 100);
  let m, n = M.get_str x in
  printf "%s %s\n" m n;
  printf "%s\n" (M.get_formatted_str x);
  let stream = open_out "fpif_export.output" in
  M.fpif_export stream x;
  close_out stream;
  let stream = open_in Sys.argv.(1) in
  let m = M.fpif_import stream in
  printf "%s\n" (M.get_formatted_str m);
  close_in stream;
  printf "%b\n" (M.fits_int_p x);
  Gc.full_major (); (* garbage collector full major *)
